torrentinfo.py

```diff
@@ -3,36 +3,69 @@
 """
 
 from sopel import module
-import bleach
 import requests
+import lxml
 from lxml import etree
+import re
+import bleach
 
-NYAA_URL = 'https://www.nyaa.se/?page=view&tid=%s'
-
+NYAA_URL = 'https://nyaa.si/view/%s'
+ANIDEX_URL = 'https://anidex.info/torrent/%s'
 
-@module.rule('.*https?:\/\/(?:www\.)?nyaa.(?:se|eu)\/\S*(?:\?|&)tid=(\d+).*')
+@module.rule('.*https?:\/\/(?:www\.)?nyaa\.si\/(?:view|download)\/(\d+).*')
 def nyaa_info(bot, trigger):
+    parsed_url = NYAA_URL % trigger.group(1)
     try:
-        r = requests.get(url=NYAA_URL % trigger.group(1), timeout=(10.0, 4.0))
+        r = requests.get(url=parsed_url, timeout=(10.0, 4.0))
     except requests.exceptions.ConnectTimeout:
-        bot.say("Connection timed out.")
+        return bot.say("[Nyaa] Connection timed out.")
     except requests.exceptions.ConnectionError:
-        bot.say("Couldn't connect to server.")
+        return bot.say("[Nyaa] Couldn't connect to server.")
     except requests.exceptions.ReadTimeout:
-        bot.say("Server took too long to send data.")
+        return bot.say("[Nyaa] Server took too long to send data.")
     try:
         r.raise_for_status()
     except requests.exceptions.HTTPError as e:
-        bot.say("HTTP error: " + e.message)
+        return bot.say("[Nyaa] HTTP error: " + e.message)
 
     page = etree.HTML(r.content)
 
     t = {}
 
-    t['name'] = page.cssselect('td.viewtorrentname')[0]
-    t['size'] = page.cssselect('#main > div.content > div > table.viewtable > tbody > tr:nth-child(6) > td.vtop')[0]
+    t['name'] = page.cssselect('meta[property="og:title"]')[0].get('content').replace(' :: Nyaa', '').encode('utf-8')
+    t['category'] = page.cssselect('meta[property="og:description"]')[0].get('content').split("|", 1)[0]
+    t['size'] = page.cssselect('meta[property="og:description"]')[0].get('content').split("|", 2)[1]
+    t['uploader'] = page.cssselect('meta[property="og:description"]')[0].get('content').split("|", 3)[2]
+    t['link'] = parsed_url
+
+    bot.say("[Nyaa] Name: {name} | {category}| Size:{size}|{uploader} | {link}".format(**t))
```

Comment:
Missing some spaces around placeholders.

Reply:
No, it's intentional, with these spaces it will show these as double spaces cause the tags themselves contain those already.

Reply:
Better to `strip()` the values fetched from the page and have absolute control over the spacing sent to IRC. (I won't double-post this, but it applies to both places.)